### PR TITLE
[DO NOT MERGE] Add AB Test for Weighted Suggested Related Links

### PR DESCRIPTION
**Pull request by chao-xian**
Adds in an AB Test for Related Links onto Case Studies and Detailed Guides. The pages that are to have the new links are hard-coded from a pre-generated list from the [related-links-recommender](https://github.com/alphagov/govuk-related-links-recommender/pull/83) for this test.

Trello card https://trello.com/c/wRj0CsIV/864-build-in-ab-feature-for-weighted-related-links

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Comment by chao-xian** [comment]
@oscarwyatt Note that [this line](https://github.com/alphagov/government-frontend/pull/2046/files#diff-8f720467cd4cfd6b9a611598b9491f0ffa8582cd2d260b733a36e16c0179bfd0R222) doesn't seem to work locally (doesn't for `master` either) so I had to hard-code in a result of `true` when testing.

**Comment by chao-xian** [comment]
Also the links that are currently in the YAML file are only test pieces of data. Need a Data Scientist to provide the full list.

**Review by oscarwyatt — COMMENTED** [review]
Only a couple of questions really, otherwise seems super. I'll try and get some clarity on the future direction of this piece of work before I request changes and/or approve

File: app/controllers/concerns/weighted_links_ab_testable.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module WeightedLinksAbTestable
+  CUSTOM_DIMENSION = 123 # TODO: Get this value from PAs
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply app/controllers/concerns/weighted_links_ab_testable.rb:2]: Correct! Admin work Zach was covering.

File: app/lib/suggested_links_builder.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def suggested_related_links
+    @content_item["links"].fetch("suggested_ordered_related_items", [])
+  end
+
+  def weighted_related_links
+    weighted_page = WeightedLinksPage.find_page_with_base_path(@content_item["base_path"])
+
+    if weighted_page
+      weighted_page.related_links.map do |link|
+        Services.content_store.content_item(link)
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply app/lib/suggested_links_builder.rb:15]: Entirely depends on how long the list is... This did occur to me but I'm not sure what the impact will be. I don't think until we get the list we can optimise for it, but at that point there are caching options we could use. There's also some caching in the gds-api-adapter as well.

**Comment by chao-xian** [comment]
Nearly a month since the last work on this. I've since left Data Labs and I don't know if that team will be doing anymore on this, but I'm closing the PR in the meantime.

**Pull request closed without merging**